patches/net/minecraft/world/item/ProjectileWeaponItem.java.patch

```diff
@@ -1,5 +1,49 @@
 --- a/net/minecraft/world/item/ProjectileWeaponItem.java
 +++ b/net/minecraft/world/item/ProjectileWeaponItem.java
+@@ -24,12 +_,43 @@
+         super(p_43009_);
+     }
+ 
++    /**
++     * @deprecated Use ItemStack sensitive version {@link ProjectileWeaponItem#getSupportedHeldProjectiles(ItemStack)}
++     */
++    @Deprecated
+     public Predicate<ItemStack> getSupportedHeldProjectiles() {
+         return this.getAllSupportedProjectiles();
+     }
+ 
++    /**
++     * If the allowed projectile is weapon stack dependent, return {@code ItemStack.EMPTY}
++     * @deprecated Use ItemStack sensitive version {@link ProjectileWeaponItem#getAllSupportedProjectiles(ItemStack)}
++     */
++    @Deprecated
+     public abstract Predicate<ItemStack> getAllSupportedProjectiles();
+ 
++    /**
++     * Override this method if the weapon stack allows special projectile that would only be used if it's in hand.
++     * The default return value is a union-predicate of {@link ProjectileWeaponItem#getAllSupportedProjectiles(ItemStack)}
++     * and {@link ProjectileWeaponItem#getSupportedHeldProjectiles()}
++     *
++     * @param stack The ProjectileWeapon stack
++     * @return A predicate that returns true for supported projectile stack in hand
++     */
++    public Predicate<ItemStack> getSupportedHeldProjectiles(ItemStack stack) {
++        return getAllSupportedProjectiles(stack).or(getSupportedHeldProjectiles());
++    }
++
++    /**
++     * Override this method if the allowed projectile is weapon stack dependent.
++     *
++     * @param stack The ProjectileWeapon stack
++     * @return A predicate that returns true for all supported projectile stack regardless of position
```

Comment:
What does position mean here?

Reply:
Position here means where the projectile stack is: in hand, in inventory, etc. This is opposed to `getSupportedHeldProjectiles` where only hand is checked.

Reply:
There must be a better way to describe that as it is a bit ambiguous rn. Could just say `supported projectile stack regardless of what slot the projectiles are in`

Reply:
Yeah I figured that's unclear. Is it better if I just say `supported projectile stack`? I think we could expect people who are overriding `ProjectileWeaponStack` to read javadoc over `getSupportedHeldProjectile` and understand that the other method supports only projectile stacks on hand

Reply:
JavaDoc updated